src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
So if it is bad to not know if a string or byte array is returned, and always returning a string is wrong, then the only safe thing to do is always return a byte array, and have the user code determine if it's safe to convert to a string. I would be fine with that.